vivisect/analysis/elf/elfplt.py

```diff
@@ -103,18 +108,61 @@ def getGOT(vw, fileva):
                 logger.debug('Adding Imagebase: 0x%x', imgbase)
                 FGOT += imgbase
 
-            if FGOT != gotva:
+            if FGOT != gotva and None not in (FGOT, gotva):
                 logger.warning("Dynamics and Sections have different GOT entries: S:0x%x D:0x%x. using Dynamics", gotva, FGOT)
 
             # since Dynamics don't store the GOT size, just use to the end of the memory map
             mmva, mmsz, mmperm, mmname = vw.getMemoryMap(FGOT)
+            gotva = FGOT
             moffset = gotva - mmva
             gotsize = mmsz - moffset
-            gotva = FGOT
 
     vw.setFileMeta(filename, 'GOT', (gotva, gotsize))
     return gotva, gotsize
 
+def getGOTs(vw):
+    out = collections.defaultdict(list)
+
+    gotva = None
+    gotsize = None
+    for va, size, name, fname in vw.getSegments():
+        if name in ('.got.plt', '.got'):
+            flist = out[fname]
+
+            gottup = (va, size)
+            if gottup not in flist:
+                flist.append(gottup)
```

Comment:
So if we wanted to be even more terse, you could make `out` a `collections.defaultdict(set)` and then just do something like:
`out[fname].add((va, size))`

and avoid a check altogether

Assuming that a `set` value would even work as the return value. Not entirely sure of that.

Reply:
don't see why not.  i'll try it out.

Reply:
wait, you mean `set`, not `list`?  i just made it a `list`
i'll give it a try.

Reply:
so... is there a benefit to sets over lists?  i just made a minor mod and both seem to work the same.  i don't know if there are any performance reasons sets might be better than lists.